server/BusinessLogic/ProductManagement/Services/ProductService.cs

```diff
@@ -0,0 +1,108 @@
+using CoffeeShop.BusinessLogic.ProductManagement.DTOs;
+using CoffeeShop.BusinessLogic.ProductManagement.Interfaces;
+using CoffeeShop.BusinessLogic.Common.Interfaces;
+using System.ComponentModel.DataAnnotations;
+
+
+namespace CoffeeShop.BusinessLogic.ProductManagement.Services;
+
+public class ProductService : IProductService
+{
+    private readonly IProductRepository _productRepository;
+    private readonly IProductMappingService _productMappingService;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public ProductService(
+        IProductRepository productRepository,
+        IProductMappingService productMappingService,
+        IUnitOfWork unitOfWork
+    )
+    {
+        _productRepository = productRepository;
+        _productMappingService = productMappingService;
+        _unitOfWork = unitOfWork;
+    }
+
+    public async Task<ProductDTO> CreateProduct(CreateProductDTO dto)
+    {
+        if (dto.Price < 0 || dto.Stock < 0)
+        {
+            throw new ValidationException("Price and Stock must be non-negative.");
```

Comment:
`InvalidDomainValueException` and we should separate this validation into 2 validations